system-addon/lib/FaviconFeed.jsm

```diff
@@ -96,6 +96,10 @@ this.FaviconFeed = class FaviconFeed {
     return this._sitesByDomain || (this._sitesByDomain = new Promise(async resolve => {
       await this.loadCachedData();
       await this.maybeRefresh();
+      if (this._sitesByDomain.then) {
+        // If _sitesByDomain is still a Promise, no data was loaded from cache or fetch.
+        this._sitesByDomain = {};
```

Comment:
Might get tests passing, but any concern of the other properties that we attach off of this object not existing?

Reply:
Yeah, we check before using https://github.com/rlr/activity-streams/blob/853f9881b76ab7b6b1b8504d5e4a62df9244a6e2/system-addon/lib/FaviconFeed.jsm#L47